whylog/config/rule.py

```diff
@@ -19,10 +19,10 @@ def serialize(self):
             "constraints": self._constraints,
         }
 
-    def get_new_parsers(self, old_parsers):
+    def get_new_parsers(self, parser_name_generator):
         new_parsers = []
         for parser in itertools.chain([self._effect], self._causes):
-            if parser.name not in old_parsers:
+            if parser_name_generator.is_free_parser_name(parser.name):
```

Comment:
Well, when this condition is False, we won't save parser, yes? So there can be situation that user wants to save rule and it will be saved... but not completely! For example one parser from rule won't be saved. It shouldn't be like that - we should save whole rule or don't save it. User should be informed and have chance to change parser name if needed.